internal/impl/io/input_http_server.go

```diff
@@ -347,12 +347,13 @@ input:
 }
 
 // HTTTPInputMiddlewareMeta is a custom middleware type for HTTP server inputs that adds metadata to messages.
-type HTTTPInputMiddlewareMeta func(*http.Request) (map[string]any, error)
+type HTTTPInputMiddlewareMeta func(conf *service.ParsedConfig) (func(*http.Request) (map[string]any, error), error)
```

Comment:
A little cleaner if we give the constructor it's own type to match the pattern of other things like this https://pkg.go.dev/github.com/Jeffail/benthos/v3/public/service#BatchInputConstructor
```suggestion
type HTTTPInputMiddlewareMeta func(*http.Request) (map[string]any, error)
type HTTTPInputMiddlewareMetaConstructor func(conf *service.ParsedConfig) (HTTTPInputMiddlewareMeta, error)
```